lib/ui/painting.dart

```diff
@@ -4935,26 +4591,1632 @@ class PictureRecorder extends NativeFieldWrapperClass2 {
+    mul(sx, 0.0, 0.0,
+        0.0, sy, 0.0);
+  }
+
+  void rotate(double radians) {
+    mul(math.cos(radians), -math.sin(radians), 0.0,
+        math.sin(radians), math.cos(radians), 0.0);
+  }
+
+  void skew(double sx, double sy) {
+    mul(1.0, sx, 0.0,
+        sy, 1.0, 0.0);
+  }
+}
+
+class _BoundsAccumulator {
+  double _minX = Rect._giantScalar;
+  double _minY = Rect._giantScalar;
+  double _maxX = -Rect._giantScalar;
+  double _maxY = -Rect._giantScalar;
+
+  void accumulate(double x, double y) {
+    if (_minX > x)
+      _minX = x;
+    if (_minY > y)
+      _minY = y;
+    if (_maxX < x)
+      _maxX = x;
+    if (_maxY < y)
+      _maxY = y;
+  }
+
+  void accumulateBounds(Rect r, RSTransform rst) {
+    accumulateRstBounds(r.left, r.top, r.right, r.bottom, rst.scos, rst.ssin, rst.tx, rst.ty);
+  }
+
+  void accumulateRstBounds(double left, double top, double right, double bottom,
+                           double scos, double ssin, double tx, double ty) {
+    if (ssin == 0) {
+      accumulate(scos * left  + tx, scos * top    + ty);
+      accumulate(scos * right + tx, scos * bottom + ty);
+      return;
+    } else if (scos == 0) {
+      accumulate(-ssin * top    + tx, ssin * left  + ty);
+      accumulate(-ssin * bottom + tx, ssin * right + ty);
+      return;
+    } else {
+      accumulate(scos * left  - ssin * top    + tx, ssin * left  + scos * top    + ty);
+      accumulate(scos * right - ssin * top    + tx, ssin * right + scos * top    + ty);
+      accumulate(scos * left  - ssin * bottom + tx, ssin * left  + scos * bottom + ty);
+      accumulate(scos * right - ssin * bottom + tx, ssin * right + scos * bottom + ty);
+    }
+  }
+
+  Rect get bounds => Rect.fromLTRB(_minX, _minY, _maxX, _maxY);
+}
+
+class _DisplayListCanvas implements Canvas {
+  _DisplayListCanvas(_DisplayListPictureRecorder recorder, [Rect? cullRect])
+      : _recorder = recorder,
+        _cullRect = cullRect ?? Rect.largest,
+        _accumulator = _BoundsAccumulator(),
+        _ops = _emptyOps, _numOps = 0,
+        _data = _emptyData, _numData = 0,
+        _dataInts = _emptyInts, _dataFloats = _emptyFloats,
+        _objData = <Object>[],
+        _ctm = _MatrixTransform._identity(),
+        _ctmStack = <_MatrixTransform>[] {
+    recorder._canvas = this;
+  }
+
+  static Uint8List _emptyOps = Uint8List(0);
+  static Uint8List _cachedOps = _emptyOps;
+  static ByteData _emptyData = ByteData(0);
+  static ByteData _cachedData = _emptyData;
+  static Uint32List _emptyInts = _emptyData.buffer.asUint32List();
+  static Float32List _emptyFloats = _emptyData.buffer.asFloat32List();
+  static List<Object> _deadObjects = List<Object>.empty(growable: false);
+
+  _DisplayListPictureRecorder? _recorder;
+  Rect _cullRect;
+  _BoundsAccumulator _accumulator;
+  int _numOps;
+  Uint8List _ops;
+  int _numData;
+  ByteData _data;
+  late Uint32List _dataInts;
+  late Float32List _dataFloats;
+  List<Object> _objData;
```

Comment:
avoid abbreviations (obj).

It would probably be helpful to give this a more meaningful name if we can, or at least have a doc comment explaining what kind of objects we anticipate going in here (I assume these are Dart objects that correspond to native peers which get passed into the C++ side as pointers).

Reply:
Could this be `List<NativeFieldWrapperClass2>`?